File: app/src/main/java/com/yatatsu/passiveviewsample/ui/ApiErrorPresenter.java
package com.yatatsu.passiveviewsample.ui;


import java.net.UnknownHostException;

import retrofit.HttpException;

public class ApiErrorPresenter {

    public static String getErrorMessage(Throwable e) {
        if (e instanceof UnknownHostException) {
            return "No internet";
        }
        if (e instanceof HttpException) {
            if (((HttpException) e).code() == 404) {
                return "Not Found";
            }
        }
        return "Unknown error";
    }
}

File: app/src/main/java/com/yatatsu/passiveviewsample/ui/ApiErrorPresenter.kt
package com.yatatsu.passiveviewsample.ui


import java.net.UnknownHostException

import retrofit.HttpException

object ApiErrorPresenter {

  fun getErrorMessage(e: Throwable): String {
    if (e is UnknownHostException) {
      return "No internet"
    }
    if (e is HttpException) {
      if (e.code() == 404) {
        return "Not Found"
      }
    }
    return "Unknown error"
  }
}
